src/TTextEdit.cpp

```diff
@@ -1354,6 +1336,65 @@ void TTextEdit::copySelectionToClipboardHTML()
     return;
 }
 
+void TTextEdit::searchSelectionOnline()
+{
+    QString selectedText = getSelectedText(' ');
+    QString url = selectedText.trimmed();
```

Comment:
You're forgetting to escape the query here - so if you select any text with `?` or `&` in the middle, the query will get stuffed up.

Reply:
I had a look at `QUrl` - seems like it should escape it by default but in my tests with DuckDuckGo, the query was cut off right at the ampersand sign.

Reply:
Might want to signal to the `QUrl` creator that the input comes from "user" input, i.e. a couple of lines down do a:

    QDesktopServices::openUrl(QUrl::fromUserInput(selectedText.trimmed()));

that ought to provide the escaping and protect against _rong_ input...